cmd/thanos/sidecar.go

```diff
@@ -53,6 +55,8 @@ func registerSidecar(m map[string]setupFunc, app *kingpin.Application, name stri
 
 	uploadCompacted := cmd.Flag("shipper.upload-compacted", "[Experimental] If true sidecar will try to upload compacted blocks as well. Useful for migration purposes. Works only if compaction is disabled on Prometheus.").Default("false").Hidden().Bool()
 
+	validateProm := cmd.Flag("sidecar.validate-prom", "If true sidecar will check Prometheus' flags to ensure disabled compaction and 2h block-time.").Default("true").Hidden().Bool()
```

Comment:
This flag does not make much sense IMO. No user should be ask to mark explicitly `hey, can you double check me please?`

If only we should have `disable prom checks` flag IF our checks are controversial. (I don't think they are)